### PR TITLE
[Messenger] added support for Amazon SQS QueueUrl as DSN

### LINKED ISSUE #37305: [Messenger] Support SQS Transport with QueueUrl as DSN
**Description**  
The Amazon SQS Transport now only supports a DSN that starts with `sqs://`, 
but especially with CloudFormation the QueueUrl could be provided as an environment variable.

See https://github.com/brefphp/symfony-messenger/pull/30#issuecomment-644228492

**Example** 
```
MESSENGER_TRANSPORT_DSN=https://sqs.us-east-2.amazonaws.com/123456789012/ab1-MyQueue-A2BCDEF3GHI4
```

### PR DESCRIPTION
**Pull request by starred-gijs**
| Q             | A
| ------------- | ---
| Branch?       | master 
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | no 
| Tickets       | Fix #37305
| License       | MIT
| Doc PR        | N/A

This will allow to use an Amazon SQS QueueUrl as transport DSN

### CODE REVIEW

**Review by chalasr — COMMENTED** [review]
Should `Connection::fromDsn()` set `Connection::$queueUrl` directly when it is given a queue url (to avoid calling `getQueueUrl()` on the SQS client when we already have it)?

**Comment by starred-gijs** [comment]
> Should `Connection::fromDsn()` set `Connection::$queueUrl` directly when it is given a queue url (to avoid calling `getQueueUrl()` on the SQS client when we already have it)?

Did not consider that to be part of the purpose of this PR. I rather keep this PR as minimal as possible.

**Comment by RocKordier** [comment]
@starred-gijs for consistency reasons maybe  also this check https://github.com/symfony/symfony/blob/6d4b26d25f3dcc224d7c66eb1e3bd1665c0bea21/src/Symfony/Component/Messenger/Transport/TransportFactory.php#L48
should be adjusted.
What do you think?

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Can you please add an entry in https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Messenger/Bridge/AmazonSqs/CHANGELOG.md?

**Comment by starred-gijs** [comment]
Wasn't sure how to update the changelog. Let me know if it needs to be different.

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/CHANGELOG.md [hunk 1]
```diff
@@ -6,3 +6,8 @@ CHANGELOG
 
  * Introduced the Amazon SQS bridge.
  * Added FIFO support to the SQS transport
+
+5.1.3
```
Comment (chalasr) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/CHANGELOG.md:10]: 5.2.0 :) This should be moved to the first position also, at the beginning of the file

**Review by chalasr — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/CHANGELOG.md (hunk 1 above)
Comment (starred-gijs) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/CHANGELOG.md]: thanks!

**Review by jderusse — APPROVED** [review]

**Review by Nyholm — APPROVED** [review]
Thank you @starred-gijs 

This is a great first contribution.

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @starred-gijs. Congratulations for your first PR on Symfony!

**Pull request merged**